### ISSUE TITLE
CacheFirstSafe is no longer fetching only ?v= resources

### ISSUE THREAD

**Issue by JudahGabriel** [madskristensen/WebEssentials.AspNetCore.ServiceWorker#8]
It seems a recent update busted the cacheFirstSafe strategy: it's now caching non-versioned css and js files. For example, the Sample project is caching site.css and site.js, even though they don't have the ?v= query param.

![image](https://user-images.githubusercontent.com/312936/33135319-b408c422-cf67-11e7-83a5-9d9212353278.png)

Is this a bug, or is something else weird going on? I should expect to see *only* .html and ?v= resources inside the cache when using CacheFirstSafe, right?

**Comment by madskristensen** [comment]
This is by design. It will cache all the requests, but it will only apply a cache-first strategy to fingerprinted resources and apply a network-first strategy for everything else.

Before this change, the offline experience was completely broken when e.g. a critical CSS file wasn't fingerprinted. In other words, it wasn't safe to use

**Comment by JudahGabriel** [comment]
This is an issue for one of my projects. Is there a way to tell the service worker, don't cache *.mp3?

One of my side projects is a radio station akin to Pandora. It calls /fetch/next/song, which returns an MP3 of several MB. Running the app over the course of an hour fetches hundreds of MBs of audio. I don't want all those added to the service worker cache (right? Browsers have a limit on the cache size, I assume).

Bottom line: is there a way to say, "CacheFirstSafe, but don't cache files matching [pattern]"?

**Comment by madskristensen** [comment]
No there isn't, but there should be. Do you want to open a separate issue for that one?

**Comment by madskristensen** [comment]
I think there should be some good defaults so that .mp3, .mp4 etc. are excluded by default

**Comment by JudahGabriel** [comment]
Yeah, I can open a separate issue for that. Thanks.

**Comment by JudahGabriel** [comment]
Mads, are you sure this "cache all the things" is the right policy? As I use this library more and more, I'm beginning to doubt it's the right policy.

All of my web apps make dynamic requests like /api/widgets/getAll, which returns all widgets, or /api/widgets/id, which returns a widget by ID. There are dozens if not hundreds of such calls in my app. They clearly shouldn't be cached; they may return something new every time (in case of /widgets/getAll) or there will be an ever-growing list of them during the lifetime of the app (in the case of /widgets/id).

But with this policy of "add everything to the cache", my cache size is growing and growing. That makes me uneasy; it will eventually break my app as I outgrow the browser's quota. 

Are we sure "cache all the things" is the right policy? For my needs, I really want a "cache all the things with ?v= query param.
